crates/router/src/types/storage/payment_intent.rs

```diff
@@ -121,4 +133,57 @@ impl PaymentIntentDbExt for PaymentIntent {
             .change_context(errors::DatabaseError::Others)
             .attach_printable("Error filtering records by time range")
     }
+
+    #[instrument(skip(conn))]
+    async fn apply_filters_on_payments(
+        conn: &PgPooledConn,
+        merchant_id: &str,
+        constraints: &api::PaymentListFilterConstraints,
+    ) -> CustomResult<Vec<(Self, PaymentAttempt)>, errors::DatabaseError> {
+        let offset = constraints.offset.unwrap_or_default();
+        let mut filter = Self::table()
+            .inner_join(payment_attempt::table.on(dsl1::attempt_id.eq(dsl::active_attempt_id)))
```

Comment:
inner join seems the fastest option, but with payment attempt/intent table being the most used table, their size may grow in future and inner join on such large tables may cause a lot of memory pressure on the postgres DB.

Reply:
For now I have already used limit, and filtering result for a particular merchant_id!

Some of other ways that I can think of:
indexing: we can make attempt_id/active_attempt column indexed,
limit records and time_range: keeping a lesser limit(10) or limiting it to lesser time_range (not more than 30/60 days)